auth/standard_role_manager.cc

```diff
@@ -713,4 +714,135 @@ future<> standard_role_manager::remove_attribute(std::string_view role_name, std
                 {sstring(role_name), sstring(attribute_name)});
     }
 }
+
+future<std::vector<role_info>> standard_role_manager::get_role_info(const bool with_salted_hashes) const {
+    // This is super verbose, but it's difficult to avoid it.
+    // Unfortunately, fmt isn't able to produce `constexpr std::string_view`s.
+    const auto role_query = std::invoke([&] {
+        if (with_salted_hashes) {
+            return seastar::format("SELECT {}, {}, {}, {} FROM {}.{}",
+                meta::roles_table::role_col_name,
+                meta::roles_table::can_login_col_name,
+                meta::roles_table::is_superuser_col_name,
+                meta::roles_table::salted_hash_col_name,
+                get_auth_ks_name(_qp),
+                meta::roles_table::name);
+        }
+
+        return seastar::format("SELECT {}, {}, {} FROM {}.{}",
+                meta::roles_table::role_col_name,
+                meta::roles_table::can_login_col_name,
+                meta::roles_table::is_superuser_col_name,
+                get_auth_ks_name(_qp),
+                meta::roles_table::name);
+    });
+
+    const auto roles = co_await _qp.execute_internal(
+            role_query,
+            db::consistency_level::LOCAL_ONE, // TODO: Verify later if this is okay.
+            internal_distributed_query_state(),
+            cql3::query_processor::cache_internal::yes);
+
+    if (roles->empty()) [[unlikely]] {
+        on_internal_error(log, "No role has been found in the system");
+    }
```

Comment:
I'm not 100% sure that is an error, or that it should be considered as such.

Reply:
Agree, on_internal_error is too strict here. A user may accidentally remove all roles (they can then recover by switching to transitional auth).